app/controllers/users_controller.rb

```diff
@@ -71,6 +71,27 @@ def new
     end
   end
 
+  def show_update_email
+    @form = UpdateUserEmailForm.new({})
+  end
+
+  def update_email
+    @form = UpdateUserEmailForm.new(params["update_user_email_form"])
+    if @form.valid?
+      update_user_email(user_id: params["user_id"], email: @form.email)
+      redirect_to update_user_path
```

Comment:
does this work? I would expect it to be `update_user_path(user_id: params[:user_id])`

Reply:
We're not very consistent about this in the users_controller or anywhere else for that matter.  It seems you can use either a symbol or a string when it comes to the params coming in to a controller.

Reply:
I meant the `redirect_to` line. Rails supplies the parameter automatically, which is nice. No changes needed then...